### ISSUE TITLE
Replace pkg.yml with workflow variables

### ISSUE THREAD

**Issue by nanory** [gardenlinux/backup_package-build#5]
**What would you like to be added**:
Currently, configuration parameter regarding the package build are defined in the `pkg.yml` file within each package repo. On one hand, it allows to define package build parameters from the root directory but on the other hand it also adds redundancy because most of the information is similar between multiple package builds. Further, some information could be even gathered during runtime (e.g. verison).

As a result, I suggest to get rid of the `pkg.yml` file and use input variables of Github Action workflows to define a proper interface and use default values for the most common parameters like:
* email
* name
* distribution
* message

**Why is this needed**:
This minimizes redundancy by defining proper defaults between package builds.

**Comment by nanory** [comment]
We had a meeting with Niko and he gave us the following heads-up.
When we try to remove the `pkg.yml`, we must find an alternative solution to build Garden Linux packages locally without requiring it to use the Github pipeline.

My propose was to let the packaging pipeline parse the Github Action workflow file to get those information that we are gone try to remove from the `pkg.yml`

**Comment by alee-cd** [comment]
It seems like to me that we should create our own GitHub GardenLinux `package-build` actions module to provide input options to specific variables originally from `pkg.yml` file. Define these in each package's `.github/workflows/build.yml`.
eg:
```
- uses: gardenlinux/build-package@main
  with:
    email: "contact@gardenlinux.io"
    name: "Garden Linux Builder"
    distribution: gardenlinux
    version_suffix: gl0
    message: "Rebuild for Garden Linux."
```
So that we don't need to have `pkg.yml` anymore as we can define these in `.github/workflows/build.yml` file instead. And if not specific, the workflows should just use the default one.

Can you please confirm this is what we should do?

**Comment by alee-cd** [comment]
The version_suffix should come from `git tags` instead. So that we don't have to update the `.github/workflows/build.yml` each time if we tag a new release.

**Comment by nanory** [comment]
Hi everyone,

we had an internal discussion about this and we decided that we want to keep the `pkg.yml` but try to improve some of its flaws. 

In general, this files allows us to easily perform local builds without relying on the Github actions by easily run the build container locally on your own machine and modify some configuration for the package build accordingly.

For this, the following documentation has been created:
https://github.com/gardenlinux/package-build#local-dev

By not moving some of its configuration information to the `.github/` workflow files and Github Action parameters, we don't need to necessarily push changes to Github first in order to test them later but we could easily run them locally by not relying on those workflow configurations.

Nevertheless, we agreed that we want to tackle the redundancy issue by pre-defining some of the `pkg.yml` attributes, so that they don't need to be defined over and over again between multiple package repos like `package-iproute2`, `package-htop`, ... 

This includes especially parameters like:
-  email
-  name
-  distribution
-  message

They will be similar between all packages, so for that reason, we created the following follup-up issue:
#15 

Then, we also discussed the `version` and `version_suffix` parameter. Those could be defined by Git Tags. This however, requires one to first create a Git tag before a package could be build and released. Although this variant provides an administrator with more control about what packages and which versions are released, it also creates more effort because each time there is an update, an administrator always needs to keep track about what packages are requiring an update and then set the Git tag accordingly which increases the administration effort. And although this variant provides more control about the package build, there is already a safety net for this by not releasing ever Garden Linux version but only making them available on daily basis. 

This way, we could always jump back to an older version of an package if we think that the new version does not work properly. For this reason, we don't need this additional security measurement but we could simply build all packages with the newest version available and just add a `version_suffix` to it. This will probably be the default behavior.

For this reason, we won't use Git tags to determine versions but we will still add Git tags afterwards to freeze versions.

**Comment by nanory** [comment]
Hi everyone,

once again we had to open this issue because we noticed some major flaw about the `pkg.yml` and the version configuration in there.

Since it's up to the developer to define a version bump in the `pkg.yml`, there could be cases in which the upstream package gets updated but the version configuration in the `pkg.yml` still references the old one.

Let's take this for example. You have a package repo like `package-htop` and you created multiple patches for it for which you increased the `version_suffix` to let's say `gardenlinux4`. As a result, you get the htop version `3.3.0gardenlinux4`. If Debian now decides to update it to version `4.0.0`, you would get a `4.0.0gardenlinux4` although there were no `4.0.0gardenlinux1`, `4.0.0gardenlinux2`, ...

For this reason, we can't use this mechanism anymore but we will probably use Git Tags to determine the versioning. This will be done in ticket #9  

Nevertheless, if this is the case, the `pkg.yml` doesn't have any important information left, since a lot of its stuff can already be defined via defaults values already as stated in #15.

This bring then yet again the discussion up on the table again if there is even a reason to keep the `pkg.yml` because if most of it's information can be defined via default values and the only important information can be defined by Git Tags, then there is no reason to keep it anymore.

For that reason, we said that we will move all `pkg.yml` parameters except for the version either into the defaults part (as stated in #15) or we move it into `.github/workflows/build.yml` as parameters that could be then gathered by the `build_source` script accordingly. This script simply needs to use another source file than (`./pkg.yml` vs `.github/workflow/build.yml`).

Once we did that, we could then remove the `pkg.yml` afterwards.

This issue now deals with the removal of the `pkg.yml`.
